packages/mui-lab/src/Masonry/Masonry.js

```diff
@@ -184,6 +184,9 @@ const Masonry = React.forwardRef(function Masonry(inProps, ref) {
 
   React.useEffect(() => {
     const handleResize = () => {
+      if (!masonryRef.current || !masonryRef.current.firstChild) {
```

Comment:
When would `masonry.current` be undefined/null?

Reply:
Right before unmounting, resize observer seems to fire one last event.

1. Go to https://6gw4k.csb.app/
2. Go to page 2 (where masonry lives) and then back to page 1.

Reply:
Let me debug it, looks strange.

Reply:
Can you share the code for this?

Reply:
Can you check this: https://github.com/hbjORbj/material-ui/pull/2

Reply:
The important part is that you need to depend on the `elements` in the `handleResize`, not on the refs. Moving the observer to a ref may not even be required.